### PR TITLE
Ug encode to sentence piece text

### PR DESCRIPTION
**Pull request by ugermann**
### Description
add member function to allow encoding/decoding into SentencePieceText.

List of changes:
- adds member function to allow encoding/decoding into SentencePieceText.

### Checklist

- [ ] I have tested the code manually
- [ ] I have run regression tests
- [ ] I have read and followed CONTRIBUTING.md
- [x] I have updated CHANGELOG.md

### CODE REVIEW

File: src/data/vocab_base.h [hunk 1]
```diff
@@ -5,6 +5,25 @@
 #include "common/utils.h"
 #include "common/file_stream.h"
 
+#ifdef USE_SENTENCEPIECE
+#  if __GNUC__ >= 5
+#    pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wsuggest-override"
+#  endif
+#  include "sentencepiece/src/sentencepiece_processor.h"
```
Comment (kpu) [inline src/data/vocab_base.h:13]: So much for abstraction.  The generic vocab class shouldn't need to include sentencepiece.

File: src/data/vocab_base.h [hunk 2]
```diff
@@ -34,6 +53,20 @@ class IVocab {
 
   virtual std::string decode(const Words& sentence,
                              bool ignoreEos = true) const = 0;
+
+  virtual void encode(sentencepiece::util::min_string_view /*textSpan*/, // span of text to be encoded
+                      sentencepiece::SentencePieceText* /*encoded*/) const {
```
Comment (kpu) [inline src/data/vocab_base.h:58]: And it won't ever be because of the wrong class.  Also, now it won't compile without sentencepiece.

**Review by kpu — CHANGES_REQUESTED** [review]
Bad idea.  Breaks abstraction in a way that could have been preserved.

**Comment by abhi-agg** [comment]
@ugermann Is this PR still valid?

**Comment by ugermann** [comment]
No.

**Pull request closed without merging**